src/main/java/de/bonndan/nivio/output/map/hex/GroupAreaFactory.java

```diff
@@ -75,41 +96,42 @@
      * @param item           the current group item
      * @param items          all group items
      * @param allVertexHexes item hex mapping
+     * @param connected
      * @return the closest neighbours
      */
-    private static Set<Hex> getClosestItemsHexes(Item item, Set<Item> items, Map<Item, Hex> allVertexHexes) {
+    private static Optional<Item> getClosestItem(Item item, Set<Item> items, Map<Item, Hex> allVertexHexes, List<Item> connected) {
         Hex start = allVertexHexes.get(item);
         AtomicInteger minDist = new AtomicInteger(Integer.MAX_VALUE);
-        final Set<Hex> min = new HashSet<>();
+        AtomicReference<Item> min = new AtomicReference<>(null);
```

Comment:
Where does the necesity of the AtomicReference arise? What could (concurrently) change/read that reference?

Reply:
It's the stream below which requires a (effectively) final variable.